rust/connlib/tunnel/src/dns.rs

```diff
@@ -439,41 +421,47 @@ pub fn is_subdomain(name: &DomainName, resource: &str) -> bool {
     name == &resource
 }
 
-fn get_description(
-    name: &DomainName,
-    dns_resources: &HashMap<String, ResourceDescriptionDns>,
-) -> Option<ResourceDescriptionDns> {
-    if let Some(resource) = dns_resources.get(&name.to_string()) {
-        return Some(resource.clone());
+fn match_domain<T>(name: &DomainName, resources: &HashMap<String, T>) -> Option<T>
+where
+    T: Copy,
+{
+    // Safety: `?` is less than 254 bytes long.
+    const QUESTION_MARK: RelativeName<&'static [u8]> =
+        unsafe { RelativeName::from_octets_unchecked(b"\x01?") };
+    // Safety: `*` is less than 254 bytes long.
+    const WILDCARD: RelativeName<&'static [u8]> =
+        unsafe { RelativeName::from_octets_unchecked(b"\x01*") };
```

Comment:
I think we can use [this](https://docs.rs/domain/latest/domain/base/name/struct.RelativeName.html#method.wildcard_ref)

Reply:
It is not a `const fn` unfortunately.